### DISCUSSION TITLE
小组毕设学号部分过长会导致显示错误，且无法手动换行

### DISCUSSION THREAD

**Discussion by yatian-liu** [sjtug/SJTUThesis#703] [提问]
编译环境是交大Overleaf，小组毕设可能需要把每个组员的学号都写在封面，但因为封面文字宽度有限学号放在一行会导致显示异常，手动加\\又会导致编译错误，能否把id也设成和title一样可以手动用\\换行？苦b密院学生今年开始也需要用交大论文写毕设报告了，是不是只有密院的毕设是小组一起做的hhh

不换行的代码和显示效果：
![image](https://user-images.githubusercontent.com/34668597/126980529-e340f76f-b3b2-4c45-88a0-445ac3848119.png)
![image](https://user-images.githubusercontent.com/34668597/126980631-75a6b892-abe2-4382-af44-96f1d1411615.png)

用\\换行的代码和部分报错（22行是`\maketitle`）：
![image](https://user-images.githubusercontent.com/34668597/126981116-d81d80fa-6ffe-4ad5-80c3-4ad90966c59f.png)
![image](https://user-images.githubusercontent.com/34668597/126980919-2bc23915-cd2a-4c4d-aa00-532bab83689f.png)

**Comment by skyzh** [comment]
> 是不是只有密院的毕设是小组一起做的hhh

是的（

> 能否把id也设成和title一样可以手动用\换行？

可以是可以，但我不确定封面能不能容纳这么多行

**Comment by AlexaraWu** [comment]
有没有官方设计参考？直接硬塞到目前的标题页我觉得不太合适。

**Reply by yatian-liu** [reply]
抱歉前几天没看...官方设计参考暂时还没有，密院今年是第一次用这个模板，只给了一个doc，格式要求还比较模糊。

![image](https://user-images.githubusercontent.com/34668597/127803708-c033f8bf-acdf-42fa-9627-fcb8372d0620.png)
自己感觉和id这个field有关的是sjtuthesis.cls的这一段，本来以为在sjtusetup.tex里多加一层{}就可以编译了，但好像还是不行，请问一下是只能在cls里面把这里的id改成和前后类似的格式吗？

**Reply by skyzh** [reply]
Hi, 可以帮忙在这里发一份官方模版吗？

目前看下来应该是要参考 title 的方法来修改的。title 支持多行。

**Reply by AlexaraWu** [reply]
搜到一份机动的小组报告模板，就是换行的。

https://me.sjtu.edu.cn/bkjx/guanliwenjian/115.html

**Reply by AlexaraWu** [reply]
> 官方设计参考暂时还没有，密院今年是第一次用这个模板，只给了一个doc，格式要求还比较模糊。

doc 可以发来看看。

**Reply by yatian-liu** [reply]
> Hi, 可以帮忙在这里发一份官方模版吗？
> 
> 目前看下来应该是要参考 title 的方法来修改的。title 支持多行。

感谢回复！密院的模板附在这里了，不过这个对封面也没有详细格式要求，本身管理有点混乱2333 只要有一个workaround就可以了，不一定需要整合到repo里的，谢谢！
[JI_CapstoneDesign_SJTU_thesis_template_SU21.zip](https://github.com/sjtug/SJTUThesis/files/6915536/JI_CapstoneDesign_SJTU_thesis_template_SU21.zip)

**Reply by yatian-liu** [reply]
doc模板一部分还用的是旧格式要求，这个不是很重要，主要是封面的学号处理问题

**Reply by peterzheng98** [reply]
@yatian-liu doc上的学号样式是？

**Reply by yatian-liu** [reply]
> @yatian-liu doc上的学号样式是？

doc上并没有给出学号格式的具体要求，刚才试了一下把`sjtuthesis-undergraduate.ltx`里titlepage里面的id从makebox改成了更宽一点(240pt)的parbox效果还可以接受，唯一的小问题是中间换行的部分没有加上`\cline{2-2}`，这个有什么简单的办法解决吗？

**Comment by yatian-liu** [comment]
![sample_titlepage](https://user-images.githubusercontent.com/34668597/128824634-f61c9067-5daf-4f88-929f-51bc9fb74756.png)
神奇的密院官方在一周之后给出了参考学号样式，请问如何修改原有的模板能实现这个效果呢？不方便的话准备直接用word模板做封面然后拼接了🤣

**Reply by peterzheng98** [reply]
cc @skyzh

**Reply by skyzh** [reply]
周末一定，周末一定！

**Reply by skyzh** [reply]
可以考虑在模版源代码里把 title 的东西直接用在学号上

**Reply by YangLeiSX** [reply]
所以改掉了吗
